runtime/rasdump/dmpsup.c

```diff
@@ -532,10 +539,34 @@ configureDumpAgents(J9JavaVM *vm)
 	/* Process IBM_XE_COE_NAME */
 	mapDumpSettings(vm, agentOpts, &agentNum);
 	
-	/* Process -Xdump command-line options (L..R)*/
+
+	/*
+	 * Process -XX:[+-]HeapDumpOnOutOfMemoryError.
+	 */
+	disableIndex = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOHEAPDUMPONOOM, NULL);
+	enableIndex = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXHEAPDUMPONOOM, NULL);
+	processXXHeapDump = ((enableIndex >= 0) || (disableIndex >= 0));
+
+	/*
+	 * Process -Xdump command-line options (L..R).
+	 * Treat -XX:[+-]HeapDumpOnOutOfMemoryError as an alias of -Xdump.
+	 */
 	xdumpIndex = FIND_ARG_IN_VMARGS_FORWARD(OPTIONAL_LIST_MATCH, VMOPT_XDUMP, NULL);
 	while ( xdumpIndex >= 0 )
 	{
+		if (processXXHeapDump) {
+			if ((enableIndex > disableIndex) && (enableIndex < xdumpIndex)) {
+				enableDumpOnOutOfMemoryError(agentOpts, &agentNum);
+				processXXHeapDump = FALSE;
+				/* re-run with the same xdumpIndex to process the actual -Xdump option. */
+				continue;
```

Comment:
Wouldn't it be simpler to just remove the `continue` statement here and in the `else` branch (`xdumpIndex` hasn't changed and so should still be non-negative)?

Reply:
Very clever.  Done.